packages/kit/src/runtime/client/client.js

```diff
@@ -51,12 +52,10 @@ default_error_loader();
 
 /** @typedef {{ x: number, y: number }} ScrollPosition */
 /** @type {Record<number, ScrollPosition>} */
-let scroll_positions = {};
-try {
-	scroll_positions = JSON.parse(sessionStorage[SCROLL_KEY]);
-} catch {
-	// do nothing
-}
+const scroll_positions = storage.get(SCROLL_KEY) ?? {};
+
+/** @type {Record<string, any[]>} */
```

Comment:
```suggestion
/** @type {Record<number, any[]>} */
```

I see `initial_history_index` and `current_history_index` which both seem to be `number`s to be used as indexes for `snapshots`.